### PR TITLE
ES6 Reflow & Refactor. Eslint's Recommended settings.

### PR DESCRIPTION
**Pull request by darkenvy**
Moved imports into separate file (for globalization.)

Changed all vars into 'let' and 'const'. This gives editors more insight into the code and exposes hoisted variables, unused variables and informs us how variables are being used at a glance.

Reflowed the code per "eslint:recommended" & "prettier" settings. We use this at my work to automatically format code so our code reviews aren't about styling and more about execution. However, this PR will have the most line changes because of changes like: singleQuotes => doubleQuotes and 80Character new-lining.

Made a zim using this command: `bin/mwoffliner.script.js --mwUrl="http://en.wikipedia.org/" --adminEmail="darkenvy6@gmail.com" --articleList="wiki_list.lst" --withZimFullTextIndex --verbose`

### CODE REVIEW

File: lib/mwoffliner.lib.js [hunk 1]
```diff
@@ -43,59 +42,59 @@ function getParametersList() {
 
 function execute(argv) {
   /************************************/
-  /* CUSTOM VARIABLE SECTION **********/
+  /* CUSTOM letIABLE SECTION **********/
```
Comment (subbuss) [inline lib/mwoffliner.lib.js]: :-) this needs fixing.

**Comment by subbuss** [comment]
Minor: For the future, can you wrap commit messages to 80 chars as well?

File: lib/mwoffliner.lib.js [hunk 2]
```diff
@@ -1,40 +1,39 @@
 'use strict';
 
 /************************************/
-/* MODULE VARIABLE SECTION **********/
+/* MODULE letIABLE SECTION **********/
```
Comment (subbuss) [inline lib/mwoffliner.lib.js]: worth checking this and other comments / strings that got replaced incorrectly.

File: lib/mwoffliner.lib.js [hunk 3]
```diff
@@ -903,18 +902,18 @@ function execute(argv) {
...
           // the script below extracts the config with a regex executed on the page header returned from the api
           const scriptTags = domino.createDocument(headhtml['*'] + '</body></html>').getElementsByTagName('script');
           const regex = /mw\.config\.set\(\{.*?\}\);/mg;
           for (let i = 0; i < scriptTags.length; i++) {
-            if (scriptTags[i].text.includes('mw.config.set')) jsConfigVars = regex.exec(scriptTags[i].text);
+            if (scriptTags[i].text.includes('mw.config.set')) jsConfiglets = regex.exec(scriptTags[i].text);
           }
-          jsConfigVars = `(window.RLQ=window.RLQ||[]).push(function() {${jsConfigVars}});`;
-          jsConfigVars = jsConfigVars.replace('nosuchaction', 'view'); // to replace the wgAction config that is set to 'nosuchaction' from api but should be 'view'
+          jsConfiglets = `(window.RLQ=window.RLQ||[]).push(function() {${jsConfiglets}});`;
```
Comment (subbuss) [inline lib/mwoffliner.lib.js]: Please fix.  Bad rename.

**Comment by darkenvy** [comment]
Sure. I'll wrap commit messages to 80 chars.

**Comment by darkenvy** [comment]
Updated

File: lib/mwoffliner.lib.js [hunk 4]
```diff
@@ -856,65 +931,82 @@ function execute(argv) {
 
   function saveArticles(dump, finished) {
     // these vars will store the list of js and css dependencies for the article we are downloading. they are populated in storeDependencies and used in setFooter
-    let jsConfigVars = '';
+    let jsConfiglets = "";
```
Comment (subbuss) [inline lib/mwoffliner.lib.js]: You still need to fix this and review the file for any other search-replace instances that need to be undone.

Comment (kelson42) [reply lib/mwoffliner.lib.js]: @darkenvy Do you still plan to go ahead with that merge request?

Comment (darkenvy) [reply lib/mwoffliner.lib.js]: Yes. I fixed all the commends that got replaced.

Comment (subbuss) [reply lib/mwoffliner.lib.js]: But, I still see several instances of "jsConfigLets' in this file. Like the one above. Or did you forget to update the PR?

Comment (kelson42) [reply lib/mwoffliner.lib.js]: @darkenvy Do not forget please to rebase, have done a few trivials fixes meanwhile.

File: lib/mwoffliner.lib.js [hunk 5]
```diff
@@ -836,51 +325,51 @@ function execute(argv) {
...
         });
       }
     });
   }
 
-  function saveArticles(dump, finished) {
+  const saveArticles = (dump, finished) => {
     // these vars will store the list of js and css dependencies for the article we are downloading. they are populated in storeDependencies and used in setFooter
-    let jsConfigVars = '';
+    let jsConfiglets = "";
```
Comment (subbuss) [inline lib/mwoffliner.lib.js]: This needs fixing.

File: lib/mwoffliner.lib.js [hunk 6]
```diff
@@ -836,51 +325,51 @@ function execute(argv) {
...
 
     function storeDependencies(parsoidDoc, articleId, finished) {
-      const filenameRadical = env.computeFilenameRadical();
+      // const filenameRadical = env.computeFilenameRadical(); // badly scoped
 
       // the encodeURIComponent below is mandatory for languages with illegal letters for uri (like fa.wikipedia.org)
       // and the encodeURI is mandatory to encode the pipes '|'
       // but the '&' and '=' must not be encoded
-      const articleApiUrl = `${apiUrl}action=parse&format=json&page=${encodeURIComponent(articleId)}&prop=${encodeURI('modules|jsconfigvars|headhtml')}`;
+      const articleApiUrl = `${apiUrl}action=parse&format=json&page=${encodeURIComponent(articleId)}&prop=${encodeURI("modules|jsconfiglets|headhtml")}`;
```
Comment (subbuss) [inline lib/mwoffliner.lib.js]: This needs fixing.

File: lib/mwoffliner.lib.js [hunk 7]
```diff
@@ -895,26 +384,26 @@ function execute(argv) {
...
+            { type: "js", moduleList: jsDependenciesList },
+            { type: "css", moduleList: styleDependenciesList }
           ];
 
           allDependenciesWithType.forEach(({ type, moduleList }) =>
             moduleList.forEach(oneModule => downloadAndSaveModule(oneModule, type))
           );
 
-          // Saving, as a js module, the jsconfigvars that are set in the header of a wikipedia page
+          // Saving, as a js module, the jsconfiglets that are set in the header of a wikipedia page
```
Comment (subbuss) [inline lib/mwoffliner.lib.js]: Please please fix all these incorrect find-replace changes. I am just flagging a few.

Comment (kelson42) [reply lib/mwoffliner.lib.js]: @darkenvy Do you still plan to fix the last few small things reported by @subbuss or do you disagree with them? Should I do it?

**Pull request merged**